test/services/config.test.js

```diff
@@ -231,5 +232,48 @@ describe('services', () => {
         expect(saveMessage).to.contain(`${nestedConfig}${path.sep}config.json`);
       });
     });
+
+    describe('PluginConfig', () => {
+      let tempConfigDir;
+      beforeEach(() => {
+        tempConfigDir = tmp.dirSync({ unsafeCleanup: true });
+      });
+      afterEach(() => {
+        tempConfigDir.removeCallback();
+      });
+
+      test.it("loads an empty object when the plugin directory doesn't exist", () => {
+        const pluginConfig = new PluginConfig(tempConfigDir.name, 'test-plugin');
+        expect(pluginConfig.getConfig()).to.deep.equal({});
+      });
+
+      test.it('loads an existing plugin config', () => {
+        fs.mkdirSync(path.join(tempConfigDir.name, 'plugins', 'test-plugin'), { recursive: true });
+        fs.writeFileSync(
+          path.join(tempConfigDir.name, 'plugins', 'test-plugin', 'config.json'),
+          JSON.stringify({ hello: 'world' }),
+        );
+        const pluginConfig = new PluginConfig(tempConfigDir.name, 'test-plugin');
+        expect(pluginConfig.getConfig()).to.deep.equal({ hello: 'world' });
+      });
+
+      test.it("saves config to the plugin directory when it doesn't exist", () => {
+        const pluginConfig = new PluginConfig(tempConfigDir.name, 'test-plugin');
+        pluginConfig.setConfig({ foo: 'bar' });
+        expect(pluginConfig.getConfig()).to.deep.equal({ foo: 'bar' });
+      });
+
+      test.it('overwrites config when it already exists', () => {
+        fs.mkdirSync(path.join(tempConfigDir.name, 'plugins', 'test-plugin'), { recursive: true });
+        fs.writeFileSync(
```

Comment:
Prefer the tests not do any `fs.*` operations as it's an implementation detail. Better to set the config twice and verify only the second set of data is returned in the get,